docs/content/guides/integrations/writing-a-multi-asset-decorator-integration.mdx

```diff
@@ -0,0 +1,393 @@
+---
+title: "Writing a multi-asset decorator integration | Dagster Docs"
+description: "This guide walks through how to write a decorator based multi-asset integration."
+---
+
+# Writing a multi-asset decorator integration
+
+When working in the Dagster ecosystem, you may have noticed that decorators are frequently used. For example, assets, jobs, and ops use decorators. If you have a service that produces many assets, it's possible to define it as a multi-asset decorator — offering a consistent and intuitive developer experience to existing Dagster APIs.
+
+In the context of Dagster, decorators are helpful because they often wrap some form of processing. For example, when writing an asset, you define your processing code and then annotate the function with the <PyObject object="asset" decorator /> decorator. Then, the internal Dagster code can register the asset, assign metadata, pass in context data, or perform any other variety of operations that are required to integrate your asset code with the Dagster platform.
+
+In this guide, you'll learn how to develop a multi-asset integration for a hypothetical replication tool.
+
+---
+
+## Prerequisites
+
+To follow the steps in this guide, you'll need:
+
+- Familiarity with Dagster
+- An understanding of Python decorators — [Real Python's Primer on Python Decorators](https://realpython.com/primer-on-python-decorators/) is a fantastic introduction
+
+---
+
```

Comment:
feel free to push back, and we can address it in the jobs-to-be-done world

i wonder if we could add a "when to model your integration in this way" section, where we can explain the whys (e.g. composability), map pattern to category like "ETL", "BI" and also list example implementations like you did on the "Approaches to writing integrations"

Reply:
I do like that idea - however, let's see if we can merge this first, and then make improvements.

Much of this may be rewritten or scrapped as a part of the docs overhaul. If we can merge this, then we can consider restructuring in the docathon if that works for you.